internal/content_coding.go

```diff
@@ -133,19 +149,38 @@ type ContentEncoder interface {
 
 // GzipEncoder compresses the buffer using gzip at the default level.
 type GzipEncoder struct {
-	writer *gzip.Writer
-	buf    *bytes.Buffer
+	pwriter *pgzip.Writer
+	writer  *gzip.Writer
+	buf     *bytes.Buffer
 }
 
-func NewGzipEncoder() *GzipEncoder {
+func NewGzipEncoder(options ...EncodingOption) (*GzipEncoder, error) {
+	cfg := encoderConfig{level: pgzip.DefaultCompression}
+	for _, o := range options {
+		o(&cfg)
+	}
+
 	var buf bytes.Buffer
+	w, err := pgzip.NewWriterLevel(&buf, cfg.level)
 	return &GzipEncoder{
-		writer: gzip.NewWriter(&buf),
-		buf:    &buf,
-	}
+		pwriter: w,
+		writer:  gzip.NewWriter(&buf),
```

Comment:
Should this be NewWriterLevel and initialized before return?

Reply:
You are right. I would think that now we have the level we should use it...